### PR TITLE
alerts on no gps signal 

### PR DESCRIPTION
**Pull request by BarAgam**
Merge remote-tracking branch 'origin/feature/gpx_ux' into task/gpx_ux/alert_no_gps

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/teamagam/GimelGimel/pull/28" target="_blank">ReviewNinja

File: app/src/main/java/com/teamagam/gimelgimel/app/control/sensors/LocationFetcher.java [hunk 1]
```diff
@@ -156,23 +174,44 @@ public void requestLocationUpdates(long minSamplingFrequencyMs,
...
-    /**
-     * used for saving last known location
-     * @param context
-     * @param intent
-     */
-    public void onReceive(Context context, Intent intent) {
-        if (intent.getExtras().containsKey(LocationManager.KEY_LOCATION_CHANGED)) {
-            mLastLocation = getLocationSample(intent);
+    private void handleNewLocation(Location location) {
+        notifyGpsStatus(location);
```
Comment (teamagam1) [inline app/src/main/java/com/teamagam/gimelgimel/app/control/sensors/LocationFetcher.java:187]: why notify status each time a new location arrives? maybe only on change? (keep flag?)

Comment (teamagam1) [inline app/src/main/java/com/teamagam/gimelgimel/app/control/sensors/LocationFetcher.java:187]: לא משנה. אולי שווה לשנות את השם :)

Comment (teamagam1) [inline app/src/main/java/com/teamagam/gimelgimel/app/control/sensors/LocationFetcher.java:187]: לא משנה. אולי שווה לשנות את השם :)

File: app/src/main/java/com/teamagam/gimelgimel/app/model/entities/LocationSample.java [hunk 2]
```diff
@@ -19,7 +21,7 @@
  * (UTC time), all other
  * parameters are optional.
  */
-public class LocationSample {
+public class LocationSample implements Parcelable {
```
Comment (teamagam1) [inline app/src/main/java/com/teamagam/gimelgimel/app/model/entities/LocationSample.java:24]: I know you spent some time on it but I would use Android Studio's basic Parcelable implementation.

File: app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/data/geometries/PointGeometry.java [hunk 3]
```diff
@@ -1,13 +1,16 @@
...
+import android.os.Parcel;
+import android.os.Parcelable;
+
 import com.google.gson.annotations.SerializedName;
 
 /**
  * Simple data-object for latitude/longitude location
  */
-public class PointGeometry implements Geometry {
+public class PointGeometry implements Geometry, Parcelable {
```
Comment (teamagam1) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/data/geometries/PointGeometry.java:11]: same comment as LocationSample:

I know you spent some time on it but I would use Android Studio's basic Parcelable implementation.

File: app/src/main/java/com/teamagam/gimelgimel/app/view/LauncherActivity.java [hunk 4]
```diff
@@ -67,26 +65,30 @@ public void onCreate(Bundle savedInstanceState) {
             mLocationFetcher.addProvider(LocationFetcher.ProviderType.LOCATION_PROVIDER_NETWORK);
             mLocationFetcher.addProvider(LocationFetcher.ProviderType.LOCATION_PROVIDER_PASSIVE);
             mLocationFetcher.requestLocationUpdates(mLocationMinUpdatesMs, mLocationMinDistanceM);
-            startMainActivity();
```
Comment (teamagam1) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/LauncherActivity.java:70]: this was intentionally here.

File: app/src/main/java/com/teamagam/gimelgimel/app/view/LauncherActivity.java [hunk 5]
```diff
@@ -67,26 +65,30 @@ public void onCreate(Bundle savedInstanceState) {
...
+
+                    LocationSample loc = intent.getParcelableExtra(
+                            LocationFetcher.KEY_NEW_LOCATION_SAMPLE);
                     Log.v("Location", loc.toString());
                     GGMessagingUtils.sendUserLocationMessageAsync(loc);
                 }
             }
         });
+
+        startMainActivity();
```
Comment (teamagam1) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/LauncherActivity.java]: not so good. we will talk about it. !resolve

File: app/src/main/java/com/teamagam/gimelgimel/app/view/MainActivity.java [hunk 6]
```diff
@@ -130,11 +141,14 @@ public void run() {
...
         // Check if the GPS is enabled,
         // and ask the user if he wants to enable it
-        if(!mLocationFetcher.isGpsOn()) {
+        if (!mLocationFetcher.isGpsProviderEnabled()) {
             TurnOnGpsDialogFragment alertDialog = new TurnOnGpsDialogFragment(this);
 
             alertDialog.show();
         }
+
+        boolean shouldDisplayNoGpsSignal = !mLocationFetcher.isGpsProviderEnabled();
```
Comment (teamagam1) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/MainActivity.java]: can't you add this line to the if above ?

File: app/src/main/java/com/teamagam/gimelgimel/app/model/entities/LocationSample.java (hunk 2 above)
Comment (teamagam1) [inline app/src/main/java/com/teamagam/gimelgimel/app/model/entities/LocationSample.java:24]: !completed

File: app/src/main/java/com/teamagam/gimelgimel/app/view/LauncherActivity.java (hunk 5 above)
Comment (teamagam1) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/LauncherActivity.java]: !resolved

File: app/src/main/java/com/teamagam/gimelgimel/app/view/MainActivity.java (hunk 6 above)
Comment (teamagam1) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/MainActivity.java]: !completed

File: app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/data/geometries/PointGeometry.java (hunk 3 above)
Comment (teamagam1) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/data/geometries/PointGeometry.java:11]: !fixed

**Pull request merged**